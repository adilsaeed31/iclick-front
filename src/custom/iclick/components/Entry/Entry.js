import React, { Component } from "react";
import PropTypes from "prop-types";
import { Router } from "routes";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import GuestForm from "@reactioncommerce/components/GuestForm/v1";
import Button from "@reactioncommerce/components/Button/v1";

export default class Entry extends Component {
  static propTypes = {
    classes: PropTypes.object,
    onLoginButtonClick: PropTypes.func,
    onRegisterButtonClick: PropTypes.func,
    setEmailOnAnonymousCart: PropTypes.func,
    theme: PropTypes.object
  }

  static defaultProps = {
    onLoginButtonClick() {
      Router.pushRoute("/signin");
    },
    onRegisterButtonClick() {
      Router.pushRoute("/signup");
    },
    setEmailOnAnonymousCart() {}
  }

  render() {
    const { onLoginButtonClick, onRegisterButtonClick, setEmailOnAnonymousCart } = this.props;
    return (
      <Grid container>
        <Grid item xs={12} md={7}>
          <div>
            <Typography variant="title" gutterBottom>
              Returning Customer
            </Typography>
            <Button onClick={onLoginButtonClick} actionType="important" isFullWidth>
              Login
            </Button>
            <Button onClick={onRegisterButtonClick} actionType="secondary" isFullWidth>
              Create a new account
            </Button>
          </div>
        </Grid>
        <Grid item xs={12} md={5}>
          <div>
            <Typography variant="title" gutterBottom>
              Guest Checkout
            </Typography>
            <GuestForm onSubmit={setEmailOnAnonymousCart} />
          </div>
        </Grid>
      </Grid>
    );
  }
}
