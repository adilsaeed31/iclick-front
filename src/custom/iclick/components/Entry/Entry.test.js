import React from "react";
import renderer from "react-test-renderer";
import { ComponentsProvider } from "@reactioncommerce/components-context";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "custom/reactionTheme";
import components from "custom/componentsContext";
import Entry from "./";

test("basic snapshot", () => {
  const component = renderer.create(<ComponentsProvider value={components}>
    <MuiThemeProvider theme={theme}>
      <Entry setEmailOnAnonymousCart={() => {}} />
    </MuiThemeProvider>
  </ComponentsProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
