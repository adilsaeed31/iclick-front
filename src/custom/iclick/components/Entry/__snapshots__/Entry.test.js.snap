// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
.c2 {
  width: 1.125em;
  height: 1.125em;
}

.c0 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1999dd;
  border-color: #1999dd;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.c0:hover {
  background-color: #25b0f9;
  border-color: #25b0f9;
  color: #ffffff;
}

.c0:active {
  background-color: #057dbe;
  border-color: #057dbe;
  color: #ffffff;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-color: #5e7480;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #3c3c3c;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.c3:hover {
  background-color: #eaeff3;
  border-color: #5e7480;
  color: #3c3c3c;
}

.c3:active {
  background-color: #c9d9e1;
  border-color: #5e7480;
  color: #3c3c3c;
}

.c11 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-color: #5e7480;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #3c3c3c;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c11:hover {
  background-color: #eaeff3;
  border-color: #5e7480;
  color: #3c3c3c;
}

.c11:active {
  background-color: #c9d9e1;
  border-color: #5e7480;
  color: #3c3c3c;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c1 svg path,
.c1 svg rect {
  fill: #ffffff;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c4 svg path,
.c4 svg rect {
  fill: #3c3c3c;
}

.c5 {
  margin-bottom: 15px;
  margin-left: 0;
  margin-right: 0;
  margin-top: 15px;
}

.c6 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  color: #737373;
  display: block;
  margin-bottom: 10px;
  margin-left: 0;
  margin-right: 0;
  margin-top: 0;
}

.c9 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  display: block;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  margin-top: 10px;
}

.c7 {
  -webkit-font-smoothing: antialiased;
  background-color: #fafafa;
  border: 1px solid #cccccc;
  border-radius: 2px;
  box-sizing: border-box;
  color: #3c3c3c;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1;
  outline: none;
  padding: 8px 10px;
  position: relative;
}

.c8 {
  background-color: inherit;
  border: none;
  box-sizing: border-box;
  color: inherit;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  outline: none;
}

.c8::-webkit-input-placeholder {
  color: #cccccc;
}

.c8::-moz-placeholder {
  color: #cccccc;
}

.c8:-ms-input-placeholder {
  color: #cccccc;
}

.c8::placeholder {
  color: #cccccc;
}

.c8:-moz-read-only {
  color: #bfbfbf;
}

.c8:read-only {
  color: #bfbfbf;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 1rem 0 0 0;
}

.c10 > * {
  width: 100%;
}

@media (min-width:600px) {
  .c10 > * {
    width: auto;
  }
}

<div
  className="MuiGrid-container-4"
>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-7-67"
  >
    <div
      className="SkEntry-loginWrapper-1"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-title-107 MuiTypography-gutterBottom-119"
      >
        Returning Customer
      </h2>
      <div
        className="SkEntry-loginButton-2 c0"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div>
          Login
        </div>
        <div
          className="c1"
          style={
            Object {
              "opacity": 0,
              "paddingLeft": 0,
              "width": 0,
            }
          }
        >
          <svg
            className="c2"
            height="40px"
            style={
              Object {
                "enableBackground": "new 0 0 50 50",
              }
            }
            version="1.1"
            viewBox="0 0 50 50"
            width="40px"
            x="0px"
            y="0px"
          >
            <path
              d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
              fill="#000"
            >
              <animateTransform
                attributeName="transform"
                attributeType="xml"
                dur="1s"
                from="0 25 25"
                repeatCount="indefinite"
                to="360 25 25"
                type="rotate"
              />
            </path>
          </svg>
        </div>
      </div>
      <div
        className="SkEntry-loginButton-2 c3"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div>
          Create a new account
        </div>
        <div
          className="c4"
          style={
            Object {
              "opacity": 0,
              "paddingLeft": 0,
              "width": 0,
            }
          }
        >
          <svg
            className="c2"
            height="40px"
            style={
              Object {
                "enableBackground": "new 0 0 50 50",
              }
            }
            version="1.1"
            viewBox="0 0 50 50"
            width="40px"
            x="0px"
            y="0px"
          >
            <path
              d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
              fill="#000"
            >
              <animateTransform
                attributeName="transform"
                attributeType="xml"
                dur="1s"
                from="0 25 25"
                repeatCount="indefinite"
                to="360 25 25"
                type="rotate"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-5-65"
  >
    <div
      className="SkEntry-guestWrapper-3"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-title-107 MuiTypography-gutterBottom-119"
      >
        Guest Checkout
      </h2>
      <div
        className={null}
        style={Object {}}
      >
        <div
          className="required c5"
        >
          <label
            className="c6"
          >
            Email Address
          </label>
          <div
            className="c7"
            value=""
          >
            <input
              className="c8"
              id="email_GuestForm_1"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Email address"
              readOnly={false}
              type="email"
              value=""
            />
          </div>
          <span
            className="c9"
          >
            You will have the option to create an account and save your details after checkout.
          </span>
        </div>
        <div
          className="c10"
        >
          <div
            className="c11"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div>
              Continue as guest
            </div>
            <div
              className="c4"
              style={
                Object {
                  "opacity": 0,
                  "paddingLeft": 0,
                  "width": 0,
                }
              }
            >
              <svg
                className="c2"
                height="40px"
                style={
                  Object {
                    "enableBackground": "new 0 0 50 50",
                  }
                }
                version="1.1"
                viewBox="0 0 50 50"
                width="40px"
                x="0px"
                y="0px"
              >
                <path
                  d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
                  fill="#000"
                >
                  <animateTransform
                    attributeName="transform"
                    attributeType="xml"
                    dur="1s"
                    from="0 25 25"
                    repeatCount="indefinite"
                    to="360 25 25"
                    type="rotate"
                  />
                </path>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
